### PR TITLE
Add ng2 validation

### PR DESCRIPTION
**Pull request by frankfu03**

### CODE REVIEW

File: src/client/app/shared/form-input/datetime-input.component.ts [hunk 1]
```diff
@@ -66,6 +67,7 @@ export class DatetimeInputComponent extends AbstractInput implements OnInit, DoC
             validators.push(Validators.required);
         }
         validators.push(new DatetimeValidator());
+        validators.push(CustomValidators.maxDate(new Date(MiscUtils.getUTCDateTime())));
```
Comment (lbodor) [inline src/client/app/shared/form-input/datetime-input.component.ts]: The will reject valid dates. Current time is saved when the component is created and used later when the form is validated, i.e., at time of validation, it is no longer the current time. Implementing this check requires a validator that can itself ask for current time.

**Comment by lbodor** [comment]
These are two pull requests in one. Could we please split this PR, and merge the half that validates URLs.

File: src/client/app/shared/form-input/abstract-input.component.ts [hunk 2]
```diff
@@ -29,6 +29,7 @@ export abstract class AbstractInput implements OnInit {
                     errString += ', ';
                 }
                 let error: any = this.formControl.errors[e];
+                console.log('~~~~~ key='+e+'; error='+error);
```
Comment (lbodor) [inline src/client/app/shared/form-input/abstract-input.component.ts]: Remove debug logging please.

Comment (frankfu03) [inline src/client/app/shared/form-input/abstract-input.component.ts]: Yeah, it is done.

**Pull request merged**